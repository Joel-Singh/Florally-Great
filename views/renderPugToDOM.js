const pug = require("pug");
const { JSDOM } = require("jsdom");

module.exports = (filePath, locals) => {
  const compiledTemplate = pug.compileFile(filePath);

  const html = compiledTemplate(locals);

  const dom = new JSDOM(html);
  const document = dom.window.document;

  return document;
};
