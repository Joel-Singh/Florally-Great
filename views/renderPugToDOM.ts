import { compileFile } from "pug";
import { JSDOM } from "jsdom";

export default (absoluteFilePath, locals) => {
  const compiledTemplate = compileFile(absoluteFilePath);

  const html = compiledTemplate(locals);

  const dom = new JSDOM(html);
  const document = dom.window.document;

  return document;
};
