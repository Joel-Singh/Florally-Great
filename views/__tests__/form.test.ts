import renderPugToDOM from "../renderPugToDOM";

test.each([
  [
    "Form renders a single error",
    { errors: [{ msg: "This Error Msg Should be seen" }] },
  ],
  [
    "Form renders multiple errors",
    {
      errors: [
        { msg: "This Error Msg Should be seen 1" },
        { msg: "This Error Msg Should be seen 2" },
        { msg: "This Error Msg Should be seen 3" },
      ],
    },
  ],
])("%s", (name, locals) => {
  const form = renderPugToDOM("./views/form.pug", locals).querySelector(
    "[data-testid='errors']"
  );
  expect(form).toMatchSnapshot();
});
