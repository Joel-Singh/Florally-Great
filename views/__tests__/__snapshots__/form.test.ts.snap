// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders a single error 1`] = `
<ul
  data-testid="errors"
>
  <li>
    This Error Msg Should be seen
  </li>
</ul>
`;

exports[`Form renders multiple errors 1`] = `
<ul
  data-testid="errors"
>
  <li>
    This Error Msg Should be seen 1
  </li>
  <li>
    This Error Msg Should be seen 2
  </li>
  <li>
    This Error Msg Should be seen 3
  </li>
</ul>
`;
